ratelimit_test.go

```diff
@@ -54,13 +54,21 @@ func runTest(t *testing.T, fn func(testRunner)) {
 				return newAtomicBased(rate, opts...)
 			},
 		},
+		{
+			name: "atomic_int64",
+			constructor: func(rate int, opts ...Option) Limiter {
+				return newAtomicInt64Based(rate, opts...)
+			},
+		},
 	}
 
 	for _, tt := range impls {
 		t.Run(tt.name, func(t *testing.T) {
+			clockMock := clock.NewMock()
+			clockMock.Set(time.Now())
```

Comment:
does this affect the test in any way?

Reply:
yes, int64 based atomic rate limiter expects that `t.clock.Now().UnixNano()` will not return 0.

`atomicInt64Limiter` stores time of the next permission issue in its state and during first permission `Take` it compares current state with 0 and if it is zero then it allows take and sets `t.clock.Now().UnixNano()`

```go
if timeOfNextPermissionIssue == 0 {
	newTimeOfNextPermissionIssue = now
	if atomic.CompareAndSwapInt64(&t.state, 0, newTimeOfNextPermissionIssue) {
		break
	}
	continue
}
```

If `t.clock.Now().UnixNano()` returns 0 multiple times it can lead to multiple cycles of such "initialisation"

Reply:
Oh, this sounds important enough then to leave a comment
```suggestion
                         // Set a non-default time.Time since some limiters (int64 in particular) use the default
                         // value as "non-initialized" state.
			clockMock := clock.NewMock()
			clockMock.Set(time.Now())
```